srt-protocol/src/protocol/mod.rs

```diff
@@ -164,6 +169,14 @@ impl fmt::Debug for TimeSpan {
     }
 }
 
+impl From<Duration> for TimeSpan {
+    fn from(duration: Duration) -> TimeSpan {
+        let duration = duration.as_micros();
+        assert!(duration < i32::MAX as u128);
+        TimeSpan::from_micros(duration as i32)
```

Comment:
Changed my mind, you should use https://doc.rust-lang.org/std/primitive.i32.html#impl-TryFrom%3Ci128%3E